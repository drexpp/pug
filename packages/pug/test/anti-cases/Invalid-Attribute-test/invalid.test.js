const pug = require('../../../');

test('invalid attribute name should log warning', () => {
  const oldWarn = console.warn;
  const warnings = [];
  console.warn = warnings.push.bind(warnings);
  pug.compileFile(
    __dirname + '/invalid-character.pug'
  );
  console.warn = oldWarn;
  expect(warnings).toMatchSnapshot();
});
