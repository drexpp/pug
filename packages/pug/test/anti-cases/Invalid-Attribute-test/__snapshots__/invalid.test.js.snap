exports[`test invalid attribute name should log warning 1`] = `
Array [
  "C:\\Users\\ivan\\Desktop\\Nueva carpeta\\Pug pull request\\pug\\packages\\pug\\test\\anti-cases\\Invalid-Attribute-test/invalid-character.pug, line 1Attribute name starts with invalid character \"{\", to use special characters in attribute names, wrap the attribute name in double quotes \"\"",
]
`;
